src/DssProxyActionsCropper.sol

```diff
@@ -750,15 +748,16 @@ contract DssProxyActionsEndCharter is Common {
         bytes32 ilk = GemJoinLike(ethJoin).ilk();
         EndLike(end).cash(ilk, wad);
         uint256 wadC = _mul(wad, EndLike(end).fix(ilk)) / RAY;
-        // Flux to the proxy's UrnProxy in charter, so it can be pulled out with the managed gem join
+        // Flux to the proxy's UrnProxy in cropper, so it can be pulled out with the managed gem join
         VatLike(vat).flux(
             ilk,
             address(this),
-            CharterLike(charter).getOrCreateProxy(address(this)),
+            CropperLike(cropper).getOrCreateProxy(address(this)),
             wadC
         );
+        // TODO: do we need to tack here?
```

Comment:
Yes I believe you do. You should add a test to have the rewards follow the user during the `quit()` process.

Reply:
Are you sure it was planned for the rewards to follow? was that thought of?
There is a `fork` call in cropper's quit() (which breaks u==v) and afterwards not sure what should happen.

Reply:
OK so I think the missing piece is addressed in https://github.com/makerdao/dss-crop-join/pull/27.
Rewards do follow the vault owner after quit() is called from the proxy actions (we already have a test for that).
Rewards related to skimmed funds are not assumed to be collected (hence we use flee in cashEth and cashGem).